src/lib/sys.c

```diff
@@ -101,6 +117,18 @@ int run_command_full_params(const char *stdout_file, const char *stderr_file, ch
 		goto error_child;
 	}
 
+	if (!stdout_file) {
+		if (dup2(pipe_fds[1], STDOUT_FILENO) < 0) {
+			goto error_child;
+		}
+	}
```

Comment:
you can move the replace_fd into an else clause here to make it clear that we are only replacing the fd if stdout_file != NULL

Reply:
Done